views/index.erb

```diff
@@ -49,11 +171,42 @@
 <script language="javascript">
   console.log("I am Javascript!");
 
+  //primes solution --------------------------------------------------
   var profile_values = document.getElementsByClassName('profile-value');
   var total_words = 0;
 
   // TODO - Count the number of words in the profile.
   // hint - look at what the innerHTML property provides.
+  for(var i = 0; i < profile_values.length; i++) {
+    var innerText = profile_values[i].innerHTML;
+    total_words += innerText.split(" ").length;
+  }
 
   document.getElementById('word-count').innerHTML = total_words;
-</script>
\ No newline at end of file
+
+  //inline primes function
+  document.getElementById('primes-button').addEventListener("click", function() {
+    var number = document.getElementById('upper-number').value;
+    var resultField = document.getElementById('primes-result');
+    console.log(number);
+
+    if (isNaN(number) || number < 2) {
+      resultField.innerHTML = "Invalid number, please enter a number larger than 1"
+    }
+    else {
+      //Could not figure this bit out, I need to make some sort of server request to call the prime fucntion.
+      resultField.innerHTML = "Could not figure this bit out, I need to make some sort of server request to call the prime fucntion";
+    }
+
+  });
+
+  // Hardcoded my github url as it would not save in my profile
+  var profileItems = document.getElementsByClassName('item-value');
+
+  for(var i = 0; i < profileItems.length; i++) {
+    if(profileItems[i].innerHTML.toLowerCase() === "spidey1") {
+      profileItems[i].innerHTML="<a href='https://github.com/spidey1'>https://github.com/spidey1</a>"
```

Comment:
OK I think we're putting in some server side logic to clean off the protocol, maybe that's affecting it?